pkg/messagebroker/kafka.go

```diff
@@ -762,3 +762,8 @@ func (k *KafkaBroker) Shutdown(ctx context.Context) {
 func (k *KafkaBroker) IsClosed(_ context.Context) bool {
 	return k.isProducerClosed
 }
+
+// Flush flushes the producer buffer
+func (k *KafkaBroker) Flush(timeoutMs int) {
+	k.Producer.Flush(timeoutMs)
```

Comment:
FLush is a function of the producer client and not consumer/admin clients. Let's restrict flush operations only to producers

Reply:
All I can do is add a producer null check over here..Is that what you mean..?